### PR TITLE
week 2 assessment complete

### PR DESCRIPTION
**Pull request by JacksonPruett**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X] Career Readiness
- [X] Coding Challenges
- [X] Coding Critique
- [X] Coding Concepts
- [X] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

I had some major struggles with the pseudo code on the first assessment. I feel I have significantly improved, though I'm sure I can do better. Got any tips?

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,39 +13,101 @@
...
+const divisibleByThree = (object) => {
+    // iterate with if statement using modulo 3
+    // compare remainder of each object to 0
+    if (object.number % 3 === 0) {
+        return `${object.number} is divisible by three`
+        // return if object number is divisible by three or not
+      } else {
+        return `${object.number} is not divisible by three`
+      }
+}
```
Comment (chrisyhlee) [inline code-challenges.test.js:79]: if statements don't start iteration. I like the line by line pseudo code, I think you just need to take an extra minute to think about what you're writing and see if it's actually exactly what the code is doing or not.

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,39 +13,101 @@
...
\ No newline at end of file
+// Pseudo code:
+// input: array of random nouns
+// output: array of random nouns with capitalizations
+const wordsCapitalized = (array) => {
+    // iterate with .map
+    return array.map((string) => {
+        // using .charAt and .toUpperCase identify and capitalize the first letter in each string
+        // using .slice join the rest of the word in each string to the array
+       return string.charAt(0).toUpperCase() + string.slice(1)
```
Comment (chrisyhlee) [inline code-challenges.test.js:109]: The wording "join the rest of the word in each string to the array" is inaccurate. You are concatenating the rest of the word to the capitalized first letter of the respective word, which then gets returned in the new array created by map.

File: code-critique.js [hunk 3]
```diff
@@ -6,5 +6,31 @@
...
+
+// So to put it simply, it gets the right output (if you ignore the "undefined") but the code is sloppy. It very much says "it'll get the job done" rather than "here is your solution" and I don't like that. Now yes this is code written by an AI and not a person but it still baffles me that such a smart AI could still have such simple errors. So what have I learned? As we have been told, don't rely on artificial intelligence to do all the work. It can be used to supplment and explain in small chunks so long as you already have a general idea of the knowledge you're looking for. But if you're looking for solid solutions to things you are genuinely confused about, consult a forum.
+
+// Leaving AI out of it, I would tell a person who wrote this that "close enough" is just a friendlier way of saying something is wrong. Then I would show them the errors I found in the code.
+  
+//   console.log(findFirstVowelIndex(" Where there any vowels?"))
+
+// Output: The first vowel 'e' is at index 3.
+//         undefined
```
Comment (chrisyhlee) [inline code-critique.js:36]: Great takeaway! So important to understand that the answers AI gives you can be filled with simple mistakes like you pointed out. AI can be a great tool, but it's important to utilize it to expand your own understanding, not to just take the answer at face value.

File: interview-questions.md [hunk 4]
```diff
@@ -10,32 +10,32 @@ Type your initials here to confirm you read the instructions:
...
-Researched answer:
+Researched answer: An object is a data type that holds collections of information in the form of pairs. A class is the blueprint for objects and, like functions, can be reused and customized as needed.
 
 3. What is iteration?
 
-Your answer:
+Your answer: Iteration is to execute a set of instructions over and over again until a condition is met.
 
-Researched answer:
+Researched answer: In Javascript, loops are needed when we need to execute code over items in a code, typically elements in an array, until a condition is met. The process of repeating a set of instructions over and over again until a condition is met is called iteration.
```
Comment (chrisyhlee) [inline interview-questions.md:27]: Loops are a way to start iteration, but they are not required because there are many ways to start iteration

**Review by chrisyhlee — APPROVED** [review]
Regarding your comment about the pseudo code, I think you just need to keep getting the reps in with forming logic, and as you do this, you will understand code better, which will then get reflected in your pseudo code. Currently, the pseudo code looks good, but it's just a little inaccurate sometimes. This will improve naturally as your understanding of the code improves, which just comes with studying and practice. Overall this assessment displays Novice, but you're getting very close to Advanced Beginner, keep up the great work.

**Comment by JacksonPruett** [comment]
Damn lol I appreciate it man. Maybe next assessment will be the one!

**Comment by chrisyhlee** [comment]
> Damn lol I appreciate it man. Maybe next assessment will be the one!

You're doing a great job man, the competency levels will come with the hard work, practice, and time, just keep up the grind and you'll be fine.

**Comment by JacksonPruett** [comment]
Thanks man I’ll keep at it!

**Pull request merged**